src/interceptors/ClientRequest/createClientRequestOverride.ts

```diff
@@ -156,28 +152,31 @@ export function createClientRequestOverride(
 
       debug('request resolved body', resolvedRequestBody)
 
-      const outHeaders = this.getHeaders()
+      const outgoingHeaders = this.getHeaders()
       const resolvedRequestHeaders = Object.assign(
         {},
-        outHeaders,
+        outgoingHeaders,
         options.headers
       )
 
-      const requestHeaders = resolvedRequestHeaders
-        ? reduceHeadersObject<HeadersObject>(
-            resolvedRequestHeaders as HeadersObject,
-            (headers, name, value) => {
-              headers[name.toLowerCase()] = value
-              return headers
-            },
-            {}
-          )
-        : {}
+      const requesHeadersObject = Object.entries(
+        resolvedRequestHeaders
+      ).reduce<HeadersObject>((headersObject, [name, value]) => {
+        if (value) {
+          const corcedValue =
+            typeof value === 'number' ? value.toString() : value
+          headersObject[name.toLowerCase()] = corcedValue
```

Comment:
Without this header name normalization it creates an object containing multiple same headers with different name casing:

```js
{
  accept: '*/*',
  Accept: '*/*'
}
```

When constructing the `Headers` instance based on such input, it results in headers values' being merged despite them being the same. Raised in https://github.com/mswjs/headers-utils/issues/13. Not an issue, given we preserve the header name normalization here.

Reply:
Nice. Perhaps in a future we just renamed `corcedValue` to `coercedValue`, but not a big deal :)